pkg/crlprovider/provider.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crlprovider
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	corecrl "github.com/notaryproject/notation-core-go/revocation/crl"
+)
+
+// static concurrency-safe map to store errors while fetching CRL from CRL management provider.
+// layout:
+//
+// map["<namespace>/<name>"] = error
+var CRLErrMap sync.Map
+
+// GetCRL gets the CRL from the provider
+func GetCRL(ctx context.Context, url string, timeout time.Duration) (*corecrl.Bundle, error) {
+	if err, ok := CRLErrMap.Load(url); ok && err != nil {
```

Comment:
Suggestion:
I understand that If an error is found, the function immediately returns it to avoid unnecessary fetch attempts for known problematic URLs. But would it be beneficial to allow a limited number of retries to make it more robust in cases where transient network issues cause a temporary fetch failure?

Reply:
probably also need to consider only retry on retriable error

Reply:
Noted.

Reply:
Created a new issue for the notation-core-go and as discussed, they are likely having this enhancement in the next update.